UsersController update request

NikitaBorg2009 opened the pull request:
Updated UsersController with service

NikitaBorg2009 committed:
Updated UsersController

NikitaBorg2009 commented:
Добрый вечер, Артём. Я вот полностью исправил UsersController. Я удалил там все autowired тогда поля и все реализовал через сервисы

NikitaBorg2009 committed:
Created UserService
Created UserServiceImpl

Temik-Karpov commented on demo/src/main/java/ru/borzhemskiiNikita/demo/controllers/UsersController.java:
```diff
@@ -34,27 +30,19 @@ public String getAccess() {
     @GetMapping("/productsInTheShop")
     public String getPITS(Model model) {
-        model.addAttribute("category", category);
+        model.addAttribute("category", userService.returnCategory());
```
Для возвращения значений у нас есть спец слово Get

NikitaBorg2009 replied:
Исправил

Temik-Karpov replied:
Можно закрывать.

Temik-Karpov commented on demo/src/main/java/ru/borzhemskiiNikita/demo/controllers/UsersController.java:
```diff
@@ -164,13 +146,7 @@ public String payForProducts(Model model) {
-        }
-
-        model.addAttribute("money", money);
+        model.addAttribute("money", userService.changeMoney1());
```
В названиях не должно быть цифр и чисел

NikitaBorg2009 replied:
Исправил

Temik-Karpov replied:
Можно закрывать.

Temik-Karpov commented:
Перенести функционал из других контроллеров также в сервисы.

NikitaBorg2009 committed:
Updated UserService

NikitaBorg2009 committed:
Updated AdminController
Created AdminService
Created AdminServiceImpl

NikitaBorg2009 committed:
Updated FilterController
Created FilterService
Created FilterServiceImpl

NikitaBorg2009 committed:
Updated AdminController
Updated AuthController
Updated FilterController
Updated AdminService
Updated FilterService
Updated UserService
Updated UserServiceImpl

NikitaBorg2009 commented:
Артём Евгеньевич, добрый вечер. Все сделано, все сервисы созданы.

Temik-Karpov commented on demo/src/main/java/ru/borzhemskiiNikita/demo/services/impl/FilterServiceImpl.java:
```diff
@@ -0,0 +1,56 @@
+
+    @Override
+    public List<Product> nameFilter(String name) {
+        Stream<Product> nameFilter = products.stream().filter((product) -> {
```
Зачем лишняя переменная? Можно сразу преобразовать в List и тут же вернуть.

Temik-Karpov commented on demo/src/main/java/ru/borzhemskiiNikita/demo/controllers/AuthController.java:
```diff
@@ -43,36 +27,11 @@ public String getReg() {
-
-            users.getAllUsers().add(user1);
-
+        if (userService.authorisationProcedure(login, password).equals("admin")) {
```
Вызываешь относительно большой метод два раза, хотя мог вызвать один раз, получить результат и этот результат сравнивать с user/admin.

Temik-Karpov commented on demo/src/main/java/ru/borzhemskiiNikita/demo/controllers/UsersController.java:
```diff
@@ -124,13 +112,13 @@ public String getDelPage() {
-        if (choice.equals("yes")) {
-            if (users.payDelivery(category.getDeliveryPrice())) {
+        if (userService.checkChoice(choice)) {
+            if (userService.checkPayDelivery()) {
```
Лучше одним if обойтись.

Temik-Karpov commented on demo/src/main/java/ru/borzhemskiiNikita/demo/services/impl/AdminServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+import ru.borzhemskiiNikita.demo.models.Product;
+import ru.borzhemskiiNikita.demo.services.AdminService;
+
+public class AdminServiceImpl implements AdminService {
```
Не забываем про аннотацию Service

Temik-Karpov commented on demo/src/main/java/ru/borzhemskiiNikita/demo/services/impl/AdminServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+        if (money < 0) {
+            return false;
+        }
+        else if (!category.isDelivery()) {
```
Один if можно сделать

NikitaBorg2009 committed:
Updated AdminServiceImpl
Updated AuthController
Updated FilterServiceImpl
Updated UsersController

NikitaBorg2009 committed:
Updated FilterServiceImpl
Updated UsersController
Updated UserService
Updated UserServiceImpl
